ui/src/app/modules/environments/configuration/configuration.component.ts

```diff
@@ -0,0 +1,55 @@
+import 'rxjs/add/operator/switchMap'
+import { Observable } from 'rxjs/Rx'
+import { ActivatedRoute, Params } from '@angular/router'
+import { Component, EventEmitter, OnInit, AfterViewInit } from '@angular/core'
+import { FormGroup, FormControl, FormBuilder, Validators } from '@angular/forms'
+
+import { EnvironmentService } from '../environment.service'
+
+@Component({
+  selector: 'configuration',
+  templateUrl: './configuration.html',
+  providers: [ EnvironmentService ]
```

Comment:
Providers should be listed in the module so that components don't have to declare them.